### PR TITLE
Add Toggle Lights Script

### PR DESCRIPTION
**Pull request by TyroneMF**
- Script to toggle lights in a defined radius on/off.

### CODE REVIEW

File: Scripts/fn_toggleLights.sqf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * 0: Marker <STRING>
+ * 1: Radius <NUMBER>
+ * 2: On <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["LightsOut", 500, true] call FUNC(toggleLights);
+ * ["LightsOut2", 250, false] call tac_Scripts_fnc_toggleLights;
```
Comment (jonpas) [inline Scripts/fn_toggleLights.sqf:17]:
```suggestion
 * ["LightsOut2", 250, false] call TAC_Scripts_fnc_toggleLights;
```

File: Scripts/fn_toggleLights.sqf [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Muta, TyroneMF
+ * Switches Lights on or off in a defined radius.
+ * Call from init.sqf or Trigger.
+ *
+ * Arguments:
+ * 0: Marker <STRING>
+ * 1: Radius <NUMBER>
```
Comment (jonpas) [inline Scripts/fn_toggleLights.sqf:9]: Can't you define the radius with a marker? `(get)markerSize`

File: Scripts/fn_toggleLights.sqf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "Land_PowerPoleWooden_L_F",
+    "Land_FuelStation_01_roof_malevil_F",
+    "Land_fs_roof_F",
+    "Land_Lighthouse_03_base_F",
+    "Land_LightHouse_F",
+    "Land_Airport_Tower_F"
+];
+
+{
+    private _state = if (_on) then {"ON"} else {"OFF"};
```
Comment (jonpas) [inline Scripts/fn_toggleLights.sqf:40]:
```suggestion
    private _state = ["OFF", "ON"] select _on;
```

File: Scripts/fn_toggleLights.sqf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ * None
+ *
+ * Example:
+ * ["LightsOut", 500, true] call FUNC(toggleLights);
+ * ["LightsOut2", 250, false] call tac_Scripts_fnc_toggleLights;
+ */
+
+params ["_marker", "_radius", "_on"];
+
+_targetTypes = [
```
Comment (jonpas) [inline Scripts/fn_toggleLights.sqf:22]:
```suggestion
private _targetTypes = [
```

File: Scripts/fn_toggleLights.sqf [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "Land_fs_roof_F",
+    "Land_Lighthouse_03_base_F",
+    "Land_LightHouse_F",
+    "Land_Airport_Tower_F"
+];
+
+{
+    private _state = if (_on) then {"ON"} else {"OFF"};
+    private _markerPos = getMarkerPos _marker;
+    private _lights = _markerPos nearObjects [_x, _radius];
```
Comment (jonpas) [inline Scripts/fn_toggleLights.sqf:42]: `nearObjects` is slow, it will cause a possible stutter.

File: Scripts/fn_toggleLights.sqf [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    "Land_FuelStation_01_roof_malevil_F",
+    "Land_fs_roof_F",
+    "Land_Lighthouse_03_base_F",
+    "Land_LightHouse_F",
+    "Land_Airport_Tower_F"
+];
+
+{
+    private _state = if (_on) then {"ON"} else {"OFF"};
+    private _markerPos = getMarkerPos _marker;
```
Comment (jonpas) [inline Scripts/fn_toggleLights.sqf:41]: This and one line above should be outside of the loop, loop does not change it.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: Scripts/fn_toggleLights.sqf (hunk 2 above)
Comment (TyroneMF) [reply Scripts/fn_toggleLights.sqf:9]: You can, but it gives you an array with 2 numbers. It only expects one. The simple solution with this is probabbly best as it's just simply placing a marker and selecting how far out you want it to take effect.

File: Scripts/fn_toggleLights.sqf (hunk 5 above)
Comment (TyroneMF) [reply Scripts/fn_toggleLights.sqf:42]: Used it a few times, if it's ran from `init` nobody notices. If it's ran mid-mission with 10-15 players it's not noticable until it's over ~5km (map dependent)

Comment (jonpas) [reply Scripts/fn_toggleLights.sqf:43]: You could use `nearEntities` which accepts multiple types in one go. `nearObjects` in this script is the only reason you need 2 loops.

Comment (TyroneMF) [reply Scripts/fn_toggleLights.sqf:43]: `nearEntities` doesn't count map objects, tested in editor by placing a streetlight which was the only one that got switched off.

**Review by jonpas — APPROVED** [review]

**Pull request merged**